### DISCUSSION TITLE
Relay server not set when using command line "--config"

### DISCUSSION THREAD

**Discussion by AH-Sass** [rustdesk/rustdesk#7118] [Q&A] [answered]
### Bug Description

When invoking the command line option "--config" with an appropriate config string, ID server, API server, and Key are all properly set. Relay server is not set.

The same config string, when used with Import server config in the GUI, sets all the fields properly, including Relay server.

### How to Reproduce

Use the command line --config option with a config string that has the Relay server defined. Relay server will not be set.

### Expected Behavior

Relay server should be set.

### Operating system(s) on local side and remote side

Windows 10/11 Client

### RustDesk Version(s) on local side and remote side

Client versions 1.2.3 stable, and 1.2.4 nightly

### Screenshots

When using config string with command line option **--config**
![image](https://github.com/rustdesk/rustdesk/assets/159487458/63b9dfa3-3b67-4325-ab96-d30b35b07cac)

Using the exact same string with the GUI option **Import server config**
![image](https://github.com/rustdesk/rustdesk/assets/159487458/82472558-07ce-4221-9798-365111dccd12)

### Additional Context

This is a major barrier to rolling out the software where I work.

**Comment by GerdBernau** [comment]
For me it is also important. The more Rustdesk is used by different companies, the more conflicts arise because everyone wants to use their own ID server

**Reply by GerdBernau** [reply]
My post is unnecessary. I confused "ID server" with "Relay server"

**Reply by xlionjuan** [reply]
You could not using your own relay server without your own ID server

Another explanation, you CAN'T use your own relay with public ID server.

**Reply by ramon-flashtec** [reply]
True, but it seems that i also can' use my id-server without relay-server set :o)

**Reply by xlionjuan** [reply]
You need to point your relay server on hbbs

**Comment by rustdesk** [comment]
We do not encourage users set relay server on RustDesk client side (please set on RustDesk server side, aka, `hbbs`), so we intentionally did not add it in --config.

You do not need to set relay server explictly If relay server shares the same ip with id server, and uses default port. RustDesk will use it implicitly.

If you really need.

- For free OSS version, you can set it with `-r` (e.g. `hbbs -r <relay-server>`).

- For Pro, you can set it in web console.
https://rustdesk.com/docs/en/self-host/rustdesk-server-pro/relay/
![image](https://github.com/rustdesk/rustdesk/assets/71636191/e441bc4a-bb2c-48e3-8174-31b1d8bb9889)

**Reply by ramon-flashtec** [reply]
I disagree with this. Unless I add the Relay-Server in addition to the ID-Server, I'm not able to connect to the client with my setup.

**Reply by ramon-flashtec** [reply]
Maybe I'm blind, but I also did not find the command-line param "-r" in the docs or the source:

[https://rustdesk.com/docs/en/client/#command-line-parameters](https://rustdesk.com/docs/en/client/#command-line-parameters)

[https://github.com/rustdesk/rustdesk/blob/bdc5cded221af9697eb29aa30babce75e987fcc9/src/core_main.rs#L242](https://github.com/rustdesk/rustdesk/blob/bdc5cded221af9697eb29aa30babce75e987fcc9/src/core_main.rs#L242)

**Reply by ramon-flashtec** [reply]
Just FYI: id-server & relay-server are the same ip and I'm using Ubuntu 22.04 > Windows (multiple versions). Only works with additional Relay-Server set.

**Reply by rustdesk** [reply]
https://rustdesk.com/docs/en/self-host/rustdesk-server-oss/install/#step-2-run-hbbs-and-hbbr-on-your-server

**Reply by ramon-flashtec** [reply]
sorry, i was confused. i thought it was about the option in the client. we do not have any issues with the server, relay server is set and works like a charm. we have to conigure the relay-server field on all clients manually, which is very annoying.

**Reply by AH-Sass** [reply]
This isn't what we experience. Our clients don't relay properly if the relay server isn't set, but they relay just fine if it is set. It's the same address as the ID server, on the standard port.

**Comment by rustdesk** [comment, marked as answer]
https://github.com/rustdesk/rustdesk/commit/b85526ce54141740d6789e19cce9c15d0970fa00 nightly build please

**Comment by Transis-Felipe** [comment]
I can't configure the client settings with --config

It's only for PRO version ? I'm using the OSS

**Reply by ldumaine** [reply]
To see program output you have launch it from powershell and pipe to  Out-String ie:
ie:
 .\rustdesk-1.4.1-x86_64.exe --config blabla |  Out-String
It finally says:
"Installation and administrative privileges required!"

**Reply by ldumaine** [reply]
Finally got it to work with the following syntax.
rustdesk.exe --config "host=1.2.3.4,key=THEKEY" 
Rustdesk has to be installed before and you have to launch this command with administrative privileges.

**Reply by cdzqs** [reply]
it's work，thx

**Reply by pjbryan** [reply]
ehhh, 2026 the solution no longer works for me for some reason? i even went as far as creating a script to add the two lines needed at the end of the rustdesk2.toml xD

**Reply by rafael5597** [reply]
> ehhh, 2026 the solution no longer works for me for some reason? i even went as far as creating a script to add the two lines needed at the end of the rustdesk2.toml xD

I'm experiencing the same issue. I need to deploy the client pre-configured with both the ID and relay servers, but the executable is only picking up the ID server. I've tested both the comma-separated key-value method and the rustdesk-licensed Base64 payload in the filename, but neither sets the relay. The configuration only applies correctly if injected via CLI or typed manually into the GUI.

**Reply by pjbryan** [reply]
how did you inject the configuration using CLI? Im willing to try anything at this point

**Reply by RickGalasio** [reply]
In 2026 I noticed that NO parameters in the rustdesk.exe for Windows are being considered. On Linux it works. Neither --config, nor --get-id... Nothing. And it doesn't return anything to the terminal either.
